### DISCUSSION TITLE
Conductor-NG-UI

### DISCUSSION THREAD

**Discussion by maheshyaddanapudi** [Netflix/conductor#2185] [Show and tell]
# Conductor NG UI
## Overview

The idea is to build a single production grade Angular Web UI, to let the user interact with Conductor API, for the following
      
      • Task definitions - Create / View / Update / Delete / Requeue Task / See Polling Data

      • Workflow definitions - View / Execute / Delete

      • Workflow Executions - View / Pause / Resume / Terminate / Retry / Restart / Archive / Delete

## Motivation

To avoid the pain points of

      • Referring to documentation manually and then develop / write Request JSONs for API requests.
      
      • Better visual representation of the workflows and tasks.

## Tech / Framework used

      --> Docker Image to host the Angular Web UI. 
	  			
      --> Angular 11

            Used an existing "Architect UI" template for development

For more details please refer

https://github.com/maheshyaddanapudi/conductor-ng-ui/blob/main/README.md

**Comment by laxmikantb** [comment]
Access to XMLHttpRequest at 'http://localhost:8080/api/metadata/taskdefs' from origin 'http://localhost:4200' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

how to disable CORS for local testing?

**Comment by maheshyaddanapudi** [comment]
Hi,

It is actually not the issue at code level. Its just that Chrome or Firefox
are not allowing API calls on localhost domain.

There is a way to startup Chrome with this security disabled and the error
should go away.

If you are using mac o os, please use the following command from terminal
to start Chrome without the cors issue.

open -n -a /Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome
--args --user-data-dir="/tmp/chrome_dev_test" --disable-web-security

If your os is windows or Linux, we should be able to find similar command
for starting Chrome.

That should resolve the issue.

Thanks and Regards,
Mahesh Yaddanapudi.

**Reply by laxmikantb** [reply]
Thanks

**Reply by laxmikantb** [reply]
Hi, I am running ng-ui with security off , I am able to see the workflow.task definition, but I do not see the execution in UI, I see workflow completed in conductor. would  you please let me know, anything missing.

**Comment by maheshyaddanapudi** [comment]
Hi,

I have just tested the workflow history in UI.

1) Created a basic Healthcheck workflow
2) Executed HC workflow
3) It reflected in Workflow Executions
4) Navigated successfully to Execution Details.

PFA Screenshots.

[image: Screenshot 2021-12-21 at 7.34.01 PM.png]

[image: Screenshot 2021-12-21 at 7.33.41 PM.png]

P.S. I am running it without security too.

Note: Are you having Elasticsearch also up and running which conductor uses
to store the execution history ? If not , then the executions will not show
up in the UI.

On Fri, Dec 17, 2021 at 5:27 PM laxmikantb ***@***.***> wrote:

> Hi, I am running ng-ui with security off , I am able to see the
> workflow.task definition, but I do not see the execution in UI, I see
> workflow completed in conductor. would you please let me know, anything
> missing.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Netflix/conductor/discussions/2185#discussioncomment-1836815>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFZZDHJ2GCJ2V6EHEFMJCPDURNQI3ANCNFSM43CIKOHA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Thanks and Regards
Mahesh Yaddanapudi.
E: ***@***.***

**Comment by maheshyaddanapudi** [comment]
Please try accessing conductor workflow execution API directly in browser

http://localhost:8080/api/workflow
Or
http://<CONDUCTOR_BASE_URL>/api/workflow


Please check if anything is being returned. If no, then that's the cause
and conductor needs ES.
If the url returns entries but UI doesn't show, then we need to debug
further.

**Reply by autodidactic** [reply]
hi
 i am not able to see executions of workflows in ui here localhost:5000

i am using swagger to create wflow and task def and i am able to see those in UI

is there a redis component missing that doesnt add executions to queues?
